### PR TITLE
aws-vault: only build on linux, remove macOS build

### PR DESCRIPTION
**Pull request by mtibben**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I'm the maintainer of [aws-vault](https://github.com/99designs/aws-vault). This PR ~removes aws-vault~ removes the macOS build in order to avoid confusion and a poor user experience when using aws-vault

A formula for aws-vault was recently added in https://github.com/Homebrew/homebrew-core/pull/83093. I'm sure this was well-intentioned, however the `aws-vault` binary must be signed and notorized to avoid constant keychain prompts. Failing to sign the binary leads to a very poor experience when using aws-vault with macOS Keychain.

Signed and notarized binaries are present in the forumula in homebrew-cask

### CODE REVIEW

**Comment by SMillerDev** [comment]
I think this FAQ touches upon the request here: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap#official-vendor-taps

If we can build the software, I think we should.

**Comment by issyl0** [comment]
This was originally added as a formula not a cask solely for use with Homebrew on Linux, where casks don’t work but this software was useful for accessing AWS. The codesigning requirements don’t apply on Linux.

It was moved into this repo as standard procedure as part of the merging of the linuxbrew-core and homebrew-core repos. I know of a few third-party taps that make use of the formula on Linux as a dependency, and real Homebrew on Linux users that rely on this for authenticating with AWS (myself included).

Instead of deleting it, could we make this `depends_on :linux`? That way the cask exists for macOS but Linux users get `brew install aws-vault` working just fine? (I don’t know what happens if macOS users `brew install aws-vault` - with the `depends_on` line will it default to the cask?) I agree it’s not looking like the most seamless UX.

**Comment by mtibben** [comment]
> If we can build the software, I think we should.

Yet the way homebrew currently "builds" the software is by removing parts of the the build script, resulting in a binary that does not operate as intended.

**Comment by mtibben** [comment]
OK that sounds reasonable @issyl0, I'll update the PR

File: Formula/aws-vault.rb [hunk 1]
```diff
@@ -6,25 +6,18 @@ class AwsVault < Formula
   license "MIT"
 
   bottle do
-    sha256 cellar: :any_skip_relocation, arm64_big_sur: "cb316e4c97e1d1e3bd29fd4b3c7b1f8fa9f3c5c53716e125a32b908bf7776623"
```
Comment (SMillerDev) [inline Formula/aws-vault.rb:9]: Please don't edit the bottle block

File: Formula/aws-vault.rb [hunk 2]
```diff
@@ -6,25 +6,18 @@ class AwsVault < Formula
...
-  def install
-    # Remove this line because we don't have a certificate to code sign with
-    inreplace "Makefile",
-      "codesign --options runtime --timestamp --sign \"$(CERT_ID)\" $(INSTALL_DIR)/aws-vault || true", ""
-    os = "darwin"
-    on_linux { os = "linux" }
-    arch = Hardware::CPU.arm? ? "arm64" : "amd64"
+  # macOS binaries are purposely not built at this time as they need to be signed and notorized
+  # in order to run correctly
+  depends_on :linux
```
Comment (SMillerDev) [inline Formula/aws-vault.rb:16]:
```suggestion
  depends_on :linux # at upstreams request because without signed binaries it's less convenient to use
```

**Comment by mtibben** [comment]
OK how does that look @SMillerDev @issyl0

**Comment by carlocab** [comment]
I don't understand why this won't work with the macOS Keychain if it's not `codesign`ed.

We have other formulae (e.g. `git`'s `git-credential-osxkeychain`) that work just fine with the macOS keychain despite not being `codesign`ed. It looks to me like the `Makefile` doesn't even care if `codesign` succeeds, since it does `|| true`.

**Comment by mtibben** [comment]
For example each time an unsigned app accesses the keychain you'll get a double-prompt - one for the app, one for the keychain and one for the item. And because aws-vault needs to access the keychain multiple times during an operation, so it's possible to get 4-6 prompts per command. Another thread on this behaviour is here https://developer.apple.com/forums/thread/649081

**Comment by mtibben** [comment]
> It looks to me like the `Makefile` doesn't even care if `codesign` succeeds, since it does `|| true`

I wrote the makefile. It uses the create-dmg script when building a release, not that makefile path. The path you're looking for is optimised for development

**Comment by SMillerDev** [comment]
> For example each time an unsigned app accesses the keychain you'll get a double-prompt without the option for "Always Allow" - one for the keychain and one for the item.

I only ever get one for my git credentials. And that formula isn't signed either.

**Comment by mtibben** [comment]
> I only ever get one for my git credentials. And that formula isn't signed either.

Not my or aws-vault users experience. But PRs welcome if you have worked out a better approach

**Comment by carlocab** [comment]
> For example each time an unsigned app accesses the keychain you'll get a double-prompt without the option for "Always Allow" - one for the keychain and one for the item. And because aws-vault needs to access the keychain multiple times during an operation, it's possible to see 4-6 prompts per command. Another thread on this behaviour is here https://developer.apple.com/forums/thread/649081

Then why does `git` give you an "Always Allow" option for their credential helper?

In any case, if that's the issue then we can just document it in the `caveats` and let users decide if they'd rather bear that or install `aws-vault` through some other channel. Perhaps some users don't find this as bad as you say it is -- the formula has had a few hundred installs since it was added here, and I haven't found any complaints about it here or on your repo.

**Comment by mtibben** [comment]
You might be right

I'm basing my assumptions about the multiple prompts from work that was done a number of years ago in aws-vault. But I'm having trouble reproducing the issues from back then. Perhaps this has changed in recent macOS releases?

The only difference I can see is the wording of the prompt now. The Homebrew build says `The authenticity of "aws-vault" cannot be verified`, while the codesigned binaries do not

Screen Shot 2021-09-04 at 9 00 47 am
Screen Shot 2021-09-04 at 9 01 13 am

**Comment by mtibben** [comment]
I've been testing the Homebrew build out. I'm unfortunately experiencing double prompts again when using "Allow" instead of "Always Allow".

**Comment by carlocab** [comment]
Let's just document this in the caveats for now. If we get more complaints about this behaviour we can try to do something more.

**Comment by mtibben** [comment]
OK fair enough

**Pull request closed without merging**